### PR TITLE
test: Add SDK e2e test for transaction failures

### PR DESCRIPTION
**Pull request by RamirezAlex**
```
{
  "id": "cl4frr1pe1451kmp4xfio2rlw",
  "createdAt": "2022-06-15T15:49:39.554Z",
  "updatedAt": "2022-06-15T15:49:39.631Z",
  "amount": null,
  "commitment": "Confirmed",
  "destination": null,
  "feePayer": null,
  "mint": null,
  "referenceId": "1655308179493",
  "referenceType": "DemoPayment",
  "signature": null,
  "solanaCommitted": "2022-06-15T15:49:39.629Z",
  "solanaConfirmed": null,
  "solanaFinalized": null,
  "solanaStart": null,
  "solanaTransaction": null,
  "source": null,
  "status": "Failed",
  "webhookEventStart": null,
  "webhookEventEnd": null,
  "webhookVerifyStart": null,
  "webhookVerifyEnd": null,
  "appEnvId": "cl4by8kj80158jrp4avtc95e0",
  "errors": [
    {
      "id": "cl4frr1rj1500kmp4yeoyv9dm",
      "message": "Error: failed to send transaction: Transaction simulation failed: Error processing Instruction 1: custom program error: 0x1",
      "type": "Unknown",
      "instruction": null,
      "appTransactionId": "cl4frr1pe1451kmp4xfio2rlw"
    }
  ]
}
```

### CODE REVIEW

File: libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts [hunk 1]
```diff
@@ -211,8 +212,20 @@ export class ApiTransactionDataAccessService implements OnModuleInit {
...
+      webhookEventEnd,
+      errors,
+      mint: mint.mint.address,
+      feePayer,
+      signature,
+      solanaCommitted: new Date(),
+      solanaStart,
+      source,
+      status,
+    })
```
Comment (RamirezAlex) [inline libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts:228]: Is there a better way to do this?

File: apps/sdk-e2e/src/integration/account-kinetic-sdk-e2e.spec.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+  })
+
+  it('should throw when try to create an account that already exists', async () => {
+    const res = await sdk.createAccount({ owner: daveKeypair })
+    expect(res.signature).toBeNull()
+    expect(res.amount).toBeNull()
+    expect(res.errors.length).toBeGreaterThan(0)
+    expect(res.status).toBe(AppTransactionStatus.Failed)
+    expect(res.errors[0].message).toContain('Error: failed to send transaction') // TODO: Account already exists
```
Comment (beeman) [inline apps/sdk-e2e/src/integration/account-kinetic-sdk-e2e.spec.ts:57]: Let's make sure we capture this TODO in a story.

File: libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts [hunk 3]
```diff
@@ -145,7 +145,8 @@ export class ApiTransactionDataAccessService implements OnModuleInit {
     // Solana Transaction
     appTransaction.solanaStart = new Date()
     try {
-      appTransaction.signature = await solana.sendRawTransaction(transaction)
+      const signature = await solana.sendRawTransaction(transaction)
+      appTransaction.signature = signature
```
Comment (beeman) [inline libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts:149]: We can probably inline this again: `appTransaction.signature = await solana.sendRawTransaction(transaction)`

File: libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts [hunk 4]
```diff
@@ -211,8 +212,20 @@ export class ApiTransactionDataAccessService implements OnModuleInit {
...
+
     // Return object
-    return this.updateAppTransaction(created.id, { webhookEventEnd })
+    return this.updateAppTransaction(created.id, {
+      webhookEventEnd,
+      errors,
+      mint: mint.mint.address,
+      feePayer,
+      signature,
+      solanaCommitted: new Date(),
```
Comment (beeman) [inline libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts:224]: The issue here is that this happens at the end of the whole method call. `solanaComitted` should really be set right after we got its confirmation, not after sending webhooks etc.

I think the code as it was, was good. What was the need to add way more data into this `updateTransaction`?

If it is to handle failed Solana transactions, we probably want to return in the `catch` case right after making the Solana transaction.

**Review by beeman — CHANGES_REQUESTED** [review]

**Review by beeman — APPROVED** [review]

**Pull request merged**